### PR TITLE
Add support for Python3.8

### PR DESCRIPTION
**Pull request by iamdefinitelyahuman**
### What I did
Add support for Python 3.8

### How I did it
* Replace `Constant` python AST type with deprecated type names when generating Vyper AST nodes.
* Add python3.8 build to CI

### How to verify it
Run the tests

### Description for the changelog
* support for Python3.8

### Cute Animal Picture
image

### CODE REVIEW

**Comment by fubuloubu** [comment]
Huh, fuzzing actually caught something this time lol

**Comment by iamdefinitelyahuman** [comment]
Failing test should be fixed by https://github.com/ethereum/vyper/pull/1679

**Comment by fubuloubu** [comment]
@iamdefinitelyahuman rebase master

**Comment by fubuloubu** [comment]
I'd like to merge #1657 first and rebase on that to ensure we're playing with all the latest toys

File: vyper/ast_utils.py [hunk 1]
```diff
@@ -87,17 +76,33 @@ def parse_python_ast(source_code: str,
...
+        if isinstance(node, python_ast.Num):
+            class_name = "Num"
+        elif isinstance(node, python_ast.Str):
+            class_name = "Str"
+        elif isinstance(node, python_ast.Bytes):
+            class_name = "Bytes"
+        elif isinstance(node, python_ast.NameConstant):
+            class_name = "NameConstant"
         else:
+            class_name = node.__class__.__name__
```
Comment (jacqueswww) [inline vyper/ast_utils.py]: Not sure I understand the fix here, as far as I can tell all these types are replaced by Constant(value=) now, doesn't that mean we have to translate each constant type and port over the value attribute (e.g. `.s` for Str and `.n` for number) ?

Comment (jacqueswww) [reply vyper/ast_utils.py]: And would that mean something the replacement step would have to map something like `isinstance(node.value, str)` becomes 'vyper_ast.Str(s=node.value)`

Comment (iamdefinitelyahuman) [reply vyper/ast_utils.py]: See new commits based on our discussion :+1:

**Comment by fubuloubu** [comment]
technically, `eth-tester` doesn't support 3.8 _yet_. That could be why this is an issue.

cc @carver

**Comment by carver** [comment]
> technically, `eth-tester` (and `py-evm`) doesn't support 3.8 _yet_. That could be why this is an issue.
> 
> cc @carver

Ahh, it looks like it's the blake2b-py build that's not compatible somehow. cc @davesque

**Comment by davesque** [comment]
Spending some time on this today.

**Comment by davesque** [comment]
Just released blake2b-py v0.1.3 that includes support for python 3.8.

**Pull request merged**